llvm/lib/Transforms/InstCombine/InstCombineMulDivRem.cpp

```diff
@@ -580,15 +580,21 @@ Instruction *InstCombinerImpl::foldFPSignBitOps(BinaryOperator &I) {
 
 Instruction *InstCombinerImpl::foldPowiReassoc(BinaryOperator &I) {
   auto createPowiExpr = [](BinaryOperator &I, InstCombinerImpl &IC, Value *X,
-                           Value *Y, Value *Z) {
+                           Value *Y, Value *Z, bool UpdateUsers = true) {
```

Comment:
In the new branch , the result of  **createPowiExpr(I, *this, X, Y, NegOne, false)**  is not return directly.
so it will missing to replace the later operand in **BinaryOperator::CreateFDivFMF(NewPow, Z, &I)**

Reply:
Then can't the individual uses take care of the insert? Pull out the insertion handling to the context instead of passing this in

Reply:
Apply your comment, thanks @arsenm